### PR TITLE
fix VAULT-24372

### PR DESCRIPTION
**Pull request by marcboudreau**
This PR fixes the approach used to redact the version, addresses, and cluster name from the responses of the following endpoints:
- `sys/seal-status`
- `sys/leader`
so that the redaction happens also when a non-root namespace is specified.

The current implementation works fine in the community edition of Vault and in the enterprise version when using the root namespace, but if a different namespace is specified on the request, the redaction is not done. This is due to the fact that the current implementation is done in an http.Handler that is applicable only to the root namespace.

Instead, this change moves the redaction logic further into the specific functions (that are common to both root and non-root namespace handling) to ensure consistent handling of the redaction.

### CODE REVIEW

**Review by VioletHynes — COMMENTED** [review]
This looks good so far, but I'd love to see a test for this functionality -- to make sure we don't break this in the future (especially since it's in a pretty core code path)

**Comment by marcboudreau** [comment]
@VioletHynes I've added some tests. I tried to write tests that covered more of the request handling path, but I couldn't see how to do so. We just don't have a good framework that covers TCP listeners and their options.

File: website/data/api-docs-nav-data.json [hunk 1]
```diff
@@ -440,7 +440,12 @@
       },
       {
         "title": "<code>/sys/config/ui/custom-messages</code>",
-        "path": "system/config-ui-custom-messages"
+        "path": "system/config-ui-custom-messages",
```
Comment (VioletHynes) [inline website/data/api-docs-nav-data.json:443]: I think this slipped in by mistake :)

**Review by VioletHynes — APPROVED** [review]
Looks good! Thanks for adding tests in this area. I appreciate that it's a difficult area to test.

While I've approved, I think there's another change accidentally included in this changeset (relating to the UI banners work) that should be in a different PR.

File: website/data/api-docs-nav-data.json (hunk 1 above)
Comment (marcboudreau) [reply website/data/api-docs-nav-data.json:443]: Yeah, don't know what I was thinking there. Fixing that now.

**Pull request merged**